jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -50,9 +57,18 @@ def parse_data(app, file_obj, ext=None, data_label=None):
                 pf = rgb2gray(im)
             pf = pf[::-1, :]  # Flip it
             _parse_image(app, pf, data_label, ext=ext)
-        else:  # Assume FITS
+        elif file_obj_lower.endswith('.fits'):
```

Comment:
This broke the test using `download_file` because the file comes back as `contents` without any extension. I think we should keep the `else` fallback and assume FITS if nothing else matches.

Reply:
Unless you have a better way to figure out the filetype if the filename has no extension.

Reply:
Good point, @pllim. Do you think it would be appropriate to make a tiny PR to `astropy.utils.data` that allows a user to discover file extensions of local caches of remote data? I think I have a working example, so I may open an astropy PR today.

Reply:
I've implemented a possible workaround in b60872b. 🤞🏻

Reply:
True, all parsers are affected by this problem. If that's something we want to fix generally, should that effort get its own ticket/issue so it can get prioritized/pointed? I'm not sure it is a pressing problem yet.

Reply:
Yes, we can defer, but you need to put "assume FITS" back in `else` for this PR.

Reply:
Agreed! 👍🏻